File: src/main/java/org/bazar/chat/domain/message/Message.java
package org.bazar.chat.domain.message;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;
import org.bazar.chat.domain.DomainObject;
import org.bazar.chat.domain.chat.Chat;

import java.util.UUID;

/**
 * Jpa сущность Сообщение
 */
@Getter
@Setter
@Entity
@Table(name = "message")
public class Message extends DomainObject {
    /**
     * Чат
     */
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "chat_id", nullable = false)
    private Chat chat;

    /**
     * Текст сообщения
     */
    @Column(name = "content")
    private String content;

    /**
     * Идентификатор пользователя
     */
    @Column(name = "user_id")
    private UUID userId;

    /**
     * Признак видимости сообщения
     */
    @Column(name = "visible")
    private Boolean visible = true;

    /**
     * Сообщение, на которое является ответом
     */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reply_message_id")
    private Message replyMessage;
}

File: src/main/kotlin/org/bazar/chat/persistence/entity/Message.kt
package org.bazar.chat.persistence.entity

import jakarta.persistence.*
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size
import org.hibernate.annotations.ColumnDefault
import java.time.Instant
import java.util.*

@Entity
@Table(name = "message")
class Message(
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "chat_id", nullable = false)
    val chat: Chat,

    @Size(max = 1028)
    @NotNull
    @Column(name = "content", nullable = false, length = 1028)
    var content: String,

    @Column(name = "user_id", nullable = false)
    val userId: UUID,

    @ColumnDefault("now()")
    @Column(name = "created_at", nullable = false)
    val createdAt: Instant = Instant.now(),

    @ColumnDefault("now()")
    @Column(name = "updated_at", nullable = false)
    var updatedAt: Instant = Instant.now(),

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    val id: Long? = null
)